### ISSUE TITLE
Does not work with bower if path if `bower_components` not in path

### ISSUE THREAD

**Issue by nasfarley88** [root-project/jsroot#122]
We use jsroot through bower (see #120) and unfortunately since our setup involves copying files, minifying, etc. when the browser loads jsroot, the path does not include `bower_components/jsroot/scripts`. Since our setup doesn't give easy access to the point where the js file is loaded, it's difficult to pass it `?bower`.

(For now, we can still import it as a standalone library, but it would be nice to use jsroot through a package manager.)

**Comment by linev** [comment]
"?bower" flag only says JSROOT, that it should use standard libraries like jquery and d3 from bower installation. Otherwise JSROOT just use libraries, which are shipped together.

One can directly set `JSROOT.bower_dir` variable to root folder for bower packages. Like:

    JSROOT.bower_dir = "path/to_bower";

It can be absolute or relative  path (respective to current HTML page). 
If you set such variable before starting to use any other JSROOT functions, 
all standard libraries will be load from bower

**Comment by nasfarley88** [comment]
OK, I'll give this a go and report back.

**Comment by linev** [comment]
Hi,

Does it help?

**Comment by linev** [comment]
> Not had the chance to try yet, my wife had a baby last Thursday

Congratulation to you and your wife!

> I'll be back at work in a couple of weeks

No problem.

**Comment by nasfarley88** [comment]
Finally got to testing this.

Our setup compresses all the js files provided by bower into a single file, so unfortunately it's not possible to specify a `bower_dir` since all the js is in a single file.

As far as I can tell, other libraries solve this problem by just assuming that the required libraries are present (although this seems odd to me as a non-js developer, I might be mistaken).

**Comment by linev** [comment]
According to that [link](https://bower.io/#use-packages), "bower_components/" is default location for all scripts, installed with bower.

Seems to be, you are using other package manager on the top of bower? Is it [grunt](https://bower.io/docs/tools/#grunt)? If yes, I need to check how it works.

**Comment by nasfarley88** [comment]
We only use bower and npm as package managers. 

We use gulp-concat (https://www.npmjs.com/package/gulp-concat) to merge all the files before serving them to the user. This means that all the libraries are essentially concatenated into a large file named `base.js` and then the browser executes that.

**Comment by linev** [comment]
I create example:

https://github.com/root-project/jsroot/tree/dev/demo/gulp

It  works only with "dev" branch, which I plan to release soon as "5.2.0" version.
Now it is possible to concatenate all JSROOT sources in one big file. 
You need to provide bower.json file as in example

There is  problem with MathJax.
I did not find a reliable way to use MathJax with gulp-concat.

**Comment by linev** [comment]
I decide to remove MathJax from the list of scripts in the demo.
MathJax.js wants to loads many-many additional scripts and fails with many of them.
If TLatex object appears in JSROOT drawing, MathJax will be loaded from default location 
- now it is https://root.cern/js/mathjax/latest

Please check provided demo and try it - with current 'dev' branch.

**Comment by linev** [comment]
Now functionality available in latest JSROOT release.
You can find info here:

https://github.com/root-project/jsroot/tree/master/demo/gulp

**Comment by nasfarley88** [comment]
Great, I'll try this out today.

**Comment by nasfarley88** [comment]
This looks like it's working apart from some conflicts about MathJax. Does MathJax still need to be included in dependencies if it's being loaded manually?

**Comment by linev** [comment]
You should fully exclude MathJax as it shown in example.
I would prefer to keep it in basic bower.json file of JSROOT, 
while without creating concatenated script it works normally.

**Comment by nasfarley88** [comment]
Hmm, I have but it's still not working. I think it's a problem on my end. I'll have another look soon and see if I can resolve it and report back.

**Comment by linev** [comment]
Can you add into your gulpfile.js following line:

   console.log(bowerMain());

It should not include MathJax.js

**Comment by linev** [comment]
I closing issue - also latest 5.3.0 release works for me

**Comment by nasfarley88** [comment]
Sounds good to me, I'll raise a new issue if it's a problem.
